crates/red_knot_python_semantic/resources/mdtest/type_api.md

```diff
@@ -0,0 +1,184 @@
+# Type API
+
+This document describes the internal `red_knot` API for manipulating types and querying their
```

Comment:
nit: I'd probably call it `knot_extensions`, as it feels quite analogous to https://pypi.org/project/mypy-extensions/ and https://pypi.org/project/pyre-extensions/. (We could even consider publishing a `knot_extensions` package to PyPI... though with the current API, I'm not sure there'd be _much_ utility in doing so, since you'd likely never want to execute at runtime the snippets using this API -- they're most useful for static assertions